sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/UnwrapCastInBinaryComparisonSuite.scala

```diff
@@ -403,11 +405,40 @@ class UnwrapCastInBinaryComparisonSuite extends PlanTest with ExpressionEvalHelp
       castDate(f5) > nullLit || castDate(f6) > nullLit)
   }
 
+  test("SPARK-46069: Support unwrap timestamp type to date type") {
+    val tsLit = Literal.create(ts1, TimestampType)
```

Comment:
> 3. isStartOfDay: return true if ts == floorDate

For such an important concept, we don't have unit tests for it? The optimized comparison is different depending on this condition!